### ISSUE TITLE
Feature request: optionally return results instead of printing

### ISSUE THREAD

**Issue by Cyrik** [philoskim/debux#21]
I've been thinking that it would be great to be able to get the debug results so that it is possible to show them in different UI's, similar to re-frame-debux, but without the re-frame dependency. Other use cases would be logging or piping into REBL, reveal or portal.
I've already started a prototype to play around with designs for the interface. If you are interested in the feature I'll fork and push my changes.

**Comment by philoskim** [comment]
I have been interested in this feature as well. Please send me your pull request for this feature and I will integrate it as much as possible.

Thanks for this feature request and upcoming your pull request!!

**Comment by Cyrik** [comment]
i've been playing around with a representation of the results that has all the data in a way that can be used to display it. I'm not happy yet, but your input is very welcome

**Comment by philoskim** [comment]
Your pull request was moved to a new branch 'multiple-outputs'.

**Comment by Cyrik** [comment]
@philoskim hey, sorry I didn't get back to this for some time. Hoping to continue working on this sometime this week and try to make debux a dependency of my own lib.

**Comment by meditans** [comment]
I'm also quite interested in this, as I want to display the trace data-structure in portal. I know that some of this has been done, because I see that @Cyrik 's fork of `debux` can be used as the inner layer of `omni-trace`. How much of that work is upstreamable? If it's not, was there some agreement of the shape of the datastructure?

**Comment by Cyrik** [comment]
Sorry it took so long @meditans. My version should be mergable. I think there's a tiny bit of cleanup left to not waste memory if you're only printing.

**Comment by philoskim** [comment]
I am very sorry for my too late response. I have been so busy with my new project so I forgot to review your pull request.  Today I browse through your implementation and find some potential problems.

In my opinion, the problems are 
1) Global ut/result* atom can cause some issues in the multithreaded programming environment.
2) clog and clogn doesn't have the same features yet.

Therefore I will review your reimplementation and reimplement it in my own style as soon as possible, respecting your idea and intention.

Thank you very much and sorry for your effort and  patience!

**Comment by deejayem** [comment]
Would this also allow the result to be used in a repl?

**Comment by Cyrik** [comment]
Yes it would, although as @meditans said, my implementation is still a bit messy.
